packages/ui/components/custom/date-range-picker.tsx

```diff
@@ -0,0 +1,411 @@
+"use client";
+
+import * as React from "react";
+import type { DateRange } from "react-day-picker";
+import { differenceInDays, format, isAfter, isBefore, startOfDay, subDays, subMonths, subYears } from "date-fns";
+import { CalendarIcon, ChevronDown, ChevronLeft, ChevronRight, ChevronUp, X } from "lucide-react";
+
+import { useMediaQuery } from "../../hooks";
+import { cn } from "../../utils";
+import { Button } from "../button";
+import { Calendar } from "../calendar";
+import { Collapsible, CollapsibleContent, CollapsibleTrigger } from "../collapsible";
+import { Popover, PopoverContent, PopoverTrigger } from "../popover";
+import { Separator } from "../separator";
+
+interface DateRangePickerProps {
+  className?: string;
+  date?: DateRange;
+  onDateChange?: (date: DateRange | undefined) => void;
+  minDate?: Date;
+  maxDate?: Date;
+  maxRangeInDays?: number;
+  disableFuture?: boolean;
+  showPresets?: boolean;
+}
+
+export function DateRangePicker({
+  className,
+  date,
+  onDateChange,
```

Comment:
non-blocking: is there anything else to change? perhaps it's better to follow common input approach and just call it
```suggestion
  onChange,
```

Reply:
Only date changes so far, so onChange looks more suitable. Done.